src/connection/request.rs

```diff
@@ -5,13 +5,19 @@ use super::Message;
 pub struct Request {
     lun: LogicalUnit,
     message: Message,
+    address_and_channel: Option<(u8, u8)>,
 }
 
 impl Request {
```

Comment:
I think getting rid of `Request::lun()` and having some 

```rust
enum RequestAddress {
    Lun(LogicalUnit),
    // Hopefully replace `u8`, `u8` with some newtypes (ref other comment)
    Ipmb(LogicalUnit, u8, u8),
}

impl Request {
    pub fn address(&self, my_addr: u8) -> RequestAddress { .. }
}
```

would be good for clarity

Reply:
Makes sense. I'm struggling a bit with the names, though. Specifically I think calling the `Ipmb` variant that implies that you are requesting that `.send()` will use ipmb for the request - but it doesn't necessarily do that.

Reply:
Anything more generic will do, naturally. Perhaps `Bridged` is more accurate?

Reply:
I've left Request::lun() for now as it's used by RCMP.

Reply:
If we move `Request::lun()` to `RequestTargetAddress::lun()` I think it will make the most sense. Then you can `request.target().lun()` if you don't care about the other information (yet).

I think the names are, at the very least, descriptive enough for anyone to figure out what is going on. We can bikeshed later!